### ISSUE TITLE
[Bug]: Re-authorization issue in datasource

### ISSUE THREAD

**Issue by SreelakshmiV12** [appsmithorg/appsmith#15244]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I have created a datasource with Authentication Type: Oauth 2.0 and Grant type : Authorization code, and saved the datasource in my app. When I  tried to run an API using this datasource, it failed because access token was expired within 1 hour. So I need to re-authorize datasource in every hour to run the API. I need to know that how to re-authorize without coming to datasource.

### Steps To Reproduce

1. Create new datasource (Autheniticated API) with Authentication Type: Oauth 2.0 and Grant type : Authorization code
2. Create an API using this datasource.
3. API run without any error initially. But after every hour, access token was expired and we need to re-authorize datasource to fix the error.

### Version

Appsmith v1.7.8

**Comment by Nikhil-Nandagopal** [comment]
@SreelakshmiV12 are you working with some 3rd party API or an internal API? With most 3rd party APIs like google, I've seen that you may need to request offline access in the scopes of the API
https://developers.google.com/identity/protocols/oauth2/web-server#offline

Do re-open this if you're still facing this issue

**Comment by jessydm** [comment]
I am using zoho books as authenticated API data source on self-hosted instance of latest Appsmith. We are getting the same issue as described above.

**Comment by Nikhil-Nandagopal** [comment]
@jessydm have you tried requesting for offline access?
We would be happy to debug the issue with you

**Comment by jessydm** [comment]
Yes...the offline access was requested. Everything works fine till Token
expires and one it expires, the only way to make it work is to go back to
data source and reauthorize it.

**Comment by jessydm** [comment]
For sake of issue, I use this datasource on Zoho Books API

**Comment by NilanshBansal** [comment]
Hey @jessydm, we apologize for the inconvenience caused. 
Can you please share some more details about this issue to help us debug this better. 
You can do so by sending an email to **support@appsmith.com**

The details we require would be: 
- Are you appsmith cloud or self-hosted? 
- Which version are you using?
- What is the Authorization URL you added while configuring the datasource?
- What is the Access token URL added?
- What are the scopes added?
- Is it possible for you to share the App JSON with support@appsmith.com
